pkg/apis/core/v1beta1/helper/helper.go

```diff
@@ -187,6 +187,8 @@ func ComputeOperationType(meta metav1.ObjectMeta, lastOperation *gardencorev1bet
 		return gardencorev1beta1.LastOperationTypeMigrate
 	case meta.DeletionTimestamp != nil:
 		return gardencorev1beta1.LastOperationTypeDelete
+	case meta.Annotations[v1beta1constants.GardenerOperation] == v1beta1constants.GardenerOperationRestore:
```

Comment:
This func is also used by gardener, e.g. in the shoot controller.
Is setting the operation annotation to `restore`/`migrate` on shoots a valid scenario?

Reply:
Sorry, I seem to be missing some context here, because I couldn't really follow https://github.com/gardener/gardener/pull/4584 and this PR's description.
So feel free to just resolve this question, if it's not relevant.

Reply:
No, but it doesn't hurt either. In the shoot case, the function will not find an annotation and use the passed last operation to determine the operation type. Note that this function would only be invoked in the restore case if this operation is retried (so the status is already initialized), for the very first time this function will not be consulted.

Reply:
Ok, thanks for explaining.